pkg/controller/cronjob/utils.go

```diff
@@ -63,28 +63,19 @@ func deleteFromActiveList(sj *batchv2alpha1.CronJob, uid types.UID) {
 
 // getParentUIDFromJob extracts UID of job's parent and whether it was found
 func getParentUIDFromJob(j batchv1.Job) (types.UID, bool) {
-	creatorRefJson, found := j.ObjectMeta.Annotations[v1.CreatedByAnnotation]
-	if !found {
-		glog.V(4).Infof("Job with no created-by annotation, name %s namespace %s", j.Name, j.Namespace)
-		return types.UID(""), false
-	}
-	var sr v1.SerializedReference
-	err := json.Unmarshal([]byte(creatorRefJson), &sr)
-	if err != nil {
-		glog.V(4).Infof("Job with unparsable created-by annotation, name %s namespace %s: %v", j.Name, j.Namespace, err)
+	controllerRef := controller.GetControllerOf(&j)
+
+	if controllerRef == nil {
+		glog.V(4).Infof("Failed to obtain controller reference, name %s namespace %s", j.Name, j.Namespace)
```

Comment:
Keep the word `Job` in the log message so it's clear what kind of object we're talking about: `"Job with no ControllerRef, ..."`.